[WPE] WPE Platform: implement WPEToplevel::get_title in wayland

carlosgcampos opened the pull request:
#### 8fea187f12a146cba9ef2e8595c30b38e736413b
<pre>
[WPE] WPE Platform: implement WPEToplevel::get_title in wayland
<a href="https://bugs.webkit.org/show_bug.cgi?id=275948">https://bugs.webkit.org/show_bug.cgi?id=275948</a>

Reviewed by Nikolas Zimmermann.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(defaultTitle):
(wpeToplevelWaylandConstructed):
(wpeToplevelWaylandSetTitle):
(wpe_toplevel_wayland_class_init):
* Tools/MiniBrowser/wpe/main.cpp:
(webViewTitleChanged):
(createWebView):
(activate):

Canonical link: <a href="https://commits.webkit.org/280445@main">https://commits.webkit.org/280445@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f30e5ad1f2c0a49373983259b1c7c899861aa1e7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56576 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35903 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9049 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60187 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7017 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58703 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43526 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7208 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45825 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4906 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58606 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33753 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48824 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26686 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30534 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6156 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6020 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52491 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6428 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61870 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/486 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6533 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53083 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/488 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48884 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53038 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/418 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8423 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31730 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32817 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33901 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32563 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
```diff
@@ -647,6 +659,35 @@ static WPEBufferDMABufFormats* wpeToplevelWaylandGetPreferredDMABufFormats(WPETo
+            return;
+        }
+
+        validTitle.reset(g_strndup(title, end - title));
```
Just read the documentation of `g_strndup`.
```To copy a number of characters from a UTF-8 encoded string, use g_utf8_strncpy() instead.```

Is that relevant? I'm not a glib expert.

carlosgcampos replied:
Here were are truncating due to a limit of bytes, `g_utf8_strncpy()` takes a size in characters not bytes.

nikolaszimmermann reviewed (approved):
Looks fine.

carlosgcampos committed:
[WPE] WPE Platform: implement WPEToplevel::get_title in wayland
https://bugs.webkit.org/show_bug.cgi?id=275948

Reviewed by Nikolas Zimmermann.

* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp:
(defaultTitle):
(wpeToplevelWaylandConstructed):
(wpeToplevelWaylandSetTitle):
(wpe_toplevel_wayland_class_init):
* Tools/MiniBrowser/wpe/main.cpp:
(webViewTitleChanged):
(createWebView):
(activate):

Canonical link: https://commits.webkit.org/280445@main
